File: JavaScriptPractice/Leetcode2501.js
var longestSquareStreak = function(nums) {
    nums.sort((a,b)=>a-b);
    let theSet = new Set(nums);
    let ans=0;
    for (let num of nums)
    {
        let cur = num*num;
        let tempAns = 1;
        while(theSet.has(cur))
        {
            tempAns++;
            cur *=cur;
        }
        ans = Math.max(ans,tempAns);
    }
    return ans>1?ans:-1;

};

File: CSharpPractice/LeetcodePractice/Leetcode2501.cs
public class Solution2501 {
    public int LongestSquareStreak(int[] nums) 
    {
        Array.Sort(nums);
        HashSet<int> theSet = new HashSet<int>(nums);
        int ans=0;
        foreach(int num in nums)
        {
            long cur = num;
            int tempAns=1;
            while(true)
            {
                long next = cur*cur;
                if(next>int.MaxValue || !theSet.Contains((int)next)){break;}
                cur=next;
                tempAns++;

            }
            ans =Math.Max(ans,tempAns);
        }
        return ans>1?ans:-1;
    }
}